int/tests/conftest.py

```diff
@@ -168,3 +176,51 @@ async def http_endpoint(agent: BaseAgent):
     with suppress(asyncio.CancelledError):
         await server_task
     await agent.cleanup()
+
+
+@pytest.fixture
+async def make_did():
+    """DID factory fixture"""
+
+    async def _make_did():
+        return await create_did.asyncio(client=backchannel)
+
+    yield _make_did
+    # TODO create DID deletion method
+
+
+@pytest.fixture
+async def make_endorser_did(make_did):
+    """Endorser DID factory fixture"""
+
+    def _make_endorser_did():
+        did = make_did()
+        logger.info("Publishing DID through https://selfserve.indiciotech.io")
+        response = httpx.post(
+            url="https://selfserve.indiciotech.io/nym",
+            json={
+                "network": "testnet",
+                "did": did.result.did,
+                "verkey": did.result.verkey,
+            },
+        )
+        if response.is_error:
+            logger.info("Failed to publish DID:", response.text)
+            return
+        logger.info("DID Published")
+        return did
+
+    yield _make_endorser_did
+
+
+@pytest.fixture
+async def accept_taa(scope="session"):
```

Comment:
```suggestion
@pytest.fixture(scope="session")
async def accept_taa():
```

Scoping is put on the fixture decorator rather than as a parameter on the fixture itself